File: 283-Move-Zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        n = len(nums)
        zero_idx = 0

        for i in range(n):
            if nums[i] != 0:
                nums[i], nums[zero_idx] = nums[zero_idx], nums[i]
                zero_idx += 1

File: neetcode/283-Move-Zeroes.ts
/**
 Do not return anything, modify nums in-place instead.
 */
function moveZeroes(nums: number[]): void {
  let idx = 0

  for (let i = 0; i < nums.length; i++) {
    if (nums[i] !== 0) {
      [nums[i], nums[idx]] = [nums[idx], nums[i]]
      idx++
    }
  }
};